specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerSecurityUserConfiguration.json

```diff
@@ -0,0 +1,996 @@
+        },
+        "x-ms-examples": {
+          "Create a user rule": {
+            "$ref": "./examples/NetworkManagerUserRulePut.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "UserRules"
+        ],
+        "description": "Deletes a user rule.",
+        "operationId": "UserRules_Delete",
+        "responses": {
+          "200": {
+            "description": "Delete Succeed."
+          },
+          "204": {
+            "description": "Request successful. Resource does not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete a user rule.": {
+            "$ref": "./examples/NetworkManagerUserRuleDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SecurityConfigurationRuleListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "./networkManagerSecurityAdminConfiguration.json#/definitions/BaseRule"
+          },
+          "description": "A list of network manager security configuration rules"
+        },
+        "failedImport": {
+          "$ref": "#/definitions/FailedImport",
+          "description": "Failed Imports."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next set of results."
+        }
+      },
+      "description": "Security configuration rule list result."
+    },
+    "UserRule": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/UserRulePropertiesFormat",
+          "x-ms-client-flatten": true,
+          "description": "Indicates the properties of the security user rule"
+        },
+        "systemData": {
+          "readOnly": true,
+          "description": "The system metadata related to this resource.",
+          "$ref": "./network.json#/definitions/SystemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./networkManagerSecurityAdminConfiguration.json#/definitions/BaseRule"
+        }
+      ],
+      "description": "Network security user rule.",
+      "x-ms-discriminator-value": "UserRule"
+    },
+    "UserRulePropertiesFormat": {
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "A friendly name for the rule."
+        },
+        "description": {
+          "type": "string",
+          "description": "A description for this rule. Restricted to 140 chars."
+        },
+        "protocol": {
+          "type": "string",
+          "$ref": "./networkManagerSecurityAdminConfiguration.json#/definitions/RuleProtocol",
+          "description": "Network protocol this rule applies to."
+        },
+        "source": {
```

Comment:
sources and destinations below - since these are arrays

Reply:
make it more specific as appropriate eg sourceAddressPrefixes

Reply:
and consider abstracting all source info (address prefixes and port ranges) - in the same object instead of separately like this

Reply:
They are same properties from NSG, we want to keep these properties same as them